src/qibosoq/components/pulses.py

```diff
@@ -32,13 +33,25 @@ class Pulse:
     adc: int
     """ADC to acquire pulse back, for readout pulses."""
 
+    def get_waveform_name(self) -> Optional[str]:
+        """Return waveform name from parameters."""
+        return None
+
+    def get_waveform_style(self) -> Optional[str]:
+        """Return waveform style from parameters."""
+        return "arb"
```

Comment:
I know that you want to have methods, since below you want to make use of attributes, but you could at least make these properties
```suggestion
    @property
    def waveform_name(self) -> Optional[str]:
        """Return waveform name from parameters."""
        return None

    @property
    def waveform_style(self) -> Optional[str]:
        """Return waveform style from parameters."""
        return "arb"
```

Reply:
Moreover: how is `.waveform_name` different from `.name`?

For `.waveform_style` you could also just use `.style`, since it's unambiguous. But `.name` is not, and I'm not sure I understand the difference...

Reply:
I've changed them to properties.

"name" is, for the moment, something related to the pulse object and not the waveform... 
To be fair, it is not very useful as a parameter, but it can be nice to have when coding (name could be "drive_pulse", "higher_pulse" and so on) to have qualitative descriptions.
"waveform_name" is a signature of the waveform and depends only on other parameters (so it is also useless to send from client to server)

Reply:
Ok, so `.name` is just a description (an arbitrary label).

Reply:
Yes exactly